src/contract.ts

```diff
@@ -50,7 +50,10 @@ export async function deployContract(web3: Web3, network: string, name: string,
   const contractBuild = await getContractBuild(name);
   const contractAbi = JSON.parse(contractBuild.abi);
   const contract = new web3.eth.Contract(contractAbi);
-  return await contract.deploy({data: '0x' + contractBuild.bin, arguments: args}).send(sendOptions);
+  const deployer = await contract.deploy({data: '0x' + contractBuild.bin, arguments: args});
+  // XXX we need to sign, and infura does not accept unsigned txs
```

Comment:
You *should* be able to `send` from "unlocked" accounts and the provider will automatically ask for a signature then send the signed transaction, right?